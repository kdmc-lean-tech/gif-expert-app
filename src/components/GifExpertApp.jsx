import React, { Fragment, useState } from 'react';
import Jumbotron from 'react-bootstrap/Jumbotron';
import AddCategory from './AddCategory';
import Category from './Category';

const GifExpertApp = ({ defaultCategories=[] }) => {

  const [ categories, setCategory ] = useState(defaultCategories);

  return (
    <Fragment>
      <Jumbotron className="text-center">
        <h1>GifExpertApp</h1>
      </Jumbotron>
      <AddCategory setCategory={ setCategory }/>
      {
        categories.map((category, idx) => <Category key={ idx.toString() } category={ category }/>)
      }
    </Fragment>
  );
}

export default GifExpertApp;
