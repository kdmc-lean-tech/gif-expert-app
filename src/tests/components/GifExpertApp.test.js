import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';

describe('GifExpertApp Component Test', () => {
  const wrapper = shallow(<GifExpertApp />);
  test('should render GifExpertApp component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should show catgories list', () => {
    const categories = ['Lindsey Lohan', 'Inuyasha'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Category').length).toBe(categories.length);
  });
});
