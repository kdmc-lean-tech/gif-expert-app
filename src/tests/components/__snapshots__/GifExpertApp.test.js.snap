// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Component Test should render GifExpertApp component 1`] = `
<Fragment>
  <Jumbotron
    className="text-center"
    fluid={false}
  >
    <h1>
      GifExpertApp
    </h1>
  </Jumbotron>
  <AddCategory
    setCategory={[Function]}
  />
</Fragment>
`;

exports[`GifExpertApp Component Test should show catgories list 1`] = `
<Fragment>
  <Jumbotron
    className="text-center"
    fluid={false}
  >
    <h1>
      GifExpertApp
    </h1>
  </Jumbotron>
  <AddCategory
    setCategory={[Function]}
  />
  <Category
    category="Lindsey Lohan"
    key="0"
  />
  <Category
    category="Inuyasha"
    key="1"
  />
</Fragment>
`;
